tests/testthat/test-datatype.R

```diff
@@ -32,3 +32,37 @@ test_that("plStruct", {
 
   err_state = result(pl$Struct(bin = pl$Binary, pl$Boolean, bob = 42))
 })
+
+
+test_that("POSIXct data conversion", {
+  expect_identical(
+    pl$lit(as.POSIXct("2022-01-01"))$to_r(),
+    as.POSIXct("2022-01-01")
+  )
+
+  expect_identical(
+    pl$lit(as.POSIXct("2022-01-01", tz = "GMT"))$to_r(),
+    as.POSIXct("2022-01-01", tz = "GMT")
+  )
+
+  expect_identical(
+    pl$lit(as.POSIXct("2022-01-01", tz = "HST"))$to_r(),
+    as.POSIXct("2022-01-01", tz = "HST")
+  )
+
+  expect_identical(
+    pl$lit(as.POSIXct("2022-01-01", tz = "GMT"))$to_r(),
+    as.POSIXct("2022-01-01", tz = "GMT")
+  )
```

Comment:
this test is the same as the second one

Reply:
Testing regarding data type mapping definitely needs to be improved.
I think we can merge this into a follow-up task for now.

Reply:
As you want but adding a `# TODO` or removing the identical test takes the same time ;)

Reply:
I created an issue #592